python/semantic_kernel/connectors/memory/milvus/milvus_memory_store.py

```diff
@@ -0,0 +1,470 @@
+                data=data,
+                limit=limit,
+                search_params={"metric_type": distance_metric},
+                output_fields=["*"],
+            )[0]
+            return results, distance_metric
+        except Exception as e:
+            self._logger.debug(f"Search failed with IP, testing L2: {e}")
+            try:
+                distance_metric = distance_pairs[distance_metric.lower()]
+                results = self._client.search(
+                    collection_name=collection_name,
+                    data=data,
+                    limit=limit,
+                    search_params={"metric_type": distance_metric},
+                    output_fields=["*"],
+                )[0]
+                return results, distance_metric
+            except Exception as e:
+                self._logger.debug(f"Search failed with L2: {e}")
+                raise e
+
+    async def get_nearest_matches_async(
+        self,
+        collection_name: str,
+        embedding: ndarray,
+        limit: int,
+        min_relevance_score: float = None,
+        with_embeddings: bool = False,
+    ) -> List[Tuple[MemoryRecord, float]]:
+        """Find the nearest `limit` matches for an embedding.
+
+        Args:
+            collection_name (str): The collection to search.
+            embedding (ndarray): The embedding to search.
+            limit (int): The total results to display.
+            min_relevance_score (float, optional): Minimum distance to include. Defaults to None.
+            with_embeddings (bool, optional): Whether to include embeddings in result. Defaults to False.
+
+        Raises:
+            Exception: Missing collection
+            e: Failure to search
+
+        Returns:
+            List[Tuple[MemoryRecord, float]]: MemoryRecord and distance tuple.
+        """
+        # Check if collection exists
+        if collection_name not in self._client.list_collections():
+            self._logger.debug(
+                f"Collection {collection_name} does not exist, cannot search."
+            )
+            raise Exception(
+                f"Collection {collection_name} does not exist, cannot search."
+            )
+        # Search requests takes a list of requests.
+        if len(embedding.shape) == 1:
+            embedding = expand_dims(embedding, axis=0)
+
+        # Search based on the cached metric
+        results, search_type = self._search(
+            collection_name=collection_name,
+            data=embedding,
+            limit=limit,
+            distance_metric=self._metric_cache.get(collection_name, None) or "IP",
+        )
+
+        # Update cached metric
+        self._metric_cache[collection_name] = search_type
+
+        cleaned_results = []
+
+        if with_embeddings:
+            ids = []
+
+        # Clean up results, filter and get ids for fetch
+        for x in results:
+            if min_relevance_score is not None and x["distance"] < min_relevance_score:
```

Comment:
Thank you for raising the issue here, I was not aware.

Reply:
Update: the recently-released Milvus 2.3 ([release notes](https://github.com/milvus-io/milvus/releases/tag/v2.3.0)) has a new feature - [range search](https://milvus.io/docs/within_range.md#Conduct-a-Range-Search) - which allows doing this. This would be considerably better than the current client-side implementation.